keytool/src/main/scala/org/constellation/keytool/DeterministicKeyGen.scala

```diff
@@ -0,0 +1,43 @@
+package org.constellation.keytool
+
+import com.typesafe.scalalogging.StrictLogging
+
+import java.nio.charset.Charset
+import java.security.spec.ECGenParameterSpec
+import java.security.{SecureRandom, _}
+
+object DeterministicKeyGen extends StrictLogging {
+
+  import KeyUtils._
+
+  private val letters: List[Char] = ('a' to 'z').toList
+  // A -> 1 ; Z -> 26
+  val letterMapping: Map[Char, Int] = letters.zipWithIndex.toMap.mapValues(_ + 1)
+
+  def mapCompoundLetters(seed: String): String = {
+    seed.toLowerCase.filter(letters.contains).flatMap(letterMapping.get).mkString("")
+  }
+
+  def makeKeyPairsLetterMap(count: Int = 10, seed: String): List[KeyPair] = {
+    val seedBytes = BigInt(mapCompoundLetters(seed)).toByteArray
+    makeKeyPairs(count, seedBytes)
+  }
+
+  def makeKeyPairsUTFBytes(count: Int = 10, seed: String): List[KeyPair] = {
+    makeKeyPairs(count, seed.getBytes(Charset.forName("UTF-8")))
+  }
+
+  // TODO: SHA1PRNG requires higher byte counts for seed security -- need to expand original bytes and pad
+  def makeKeyPairs(count: Int = 10, seed: Array[Byte]): List[KeyPair] = {
+    logger.debug("Seed bytes {}", seed.size)
+    val detSecureRandom = SecureRandom.getInstance("SHA1PRNG")
```

Comment:
Nice, but def make sure it's compatible with BC's ECDSA. Also I would put this in the Wallet project, I believe keytool is deprecated. Also you'll see in Wallet that there's a BC dependency bug that non-deterministically breaks when doing BIP44. So yeah, there's that ...

Reply:
keytool isn't deprecated @buckysballs. Keytool is for everything related to private/public keys and address generation. Wallet is a CLI for transactions.

Reply:
So this code actually needs to depend on bitcoinj. I saw the existing comments around compile issues and was generating something temporary in the meantime. After more reading I'm not even sure SHA1PRNG is as secure as what bitcoinj is doing. The goal is to build a converter here between arbitrary seed bytes and mnemonics that function with bitcoin and be able to generate wallets that match existing det wallet generators.

This should only be used as a testing function until we can resolve the dependency issues. I'll address later.

Reply:
if so, please move that from src to the test package

Reply:
Moved this all to it's own test, removed most of these functions as I got BIP-44 working.